pkg/validation/test.go

```diff
@@ -879,6 +879,14 @@ func validateDNSConfig(fieldRoot string, dnsConfig []api.StepDNSConfig) (ret []e
 	return errs
 }
 
+func validateNodeArchitecture(fieldRoot string, nodeArchitecture api.NodeArchitecture) (ret []error) {
```

Comment:
It looks like this will only ever return, at most, one error. Can the function sig reflect that?